api/prometheus/v1/api.go

```diff
@@ -721,11 +723,13 @@ func (rg *RuleGroup) UnmarshalJSON(b []byte) error {
 
 	for _, rule := range v.Rules {
 		alertingRule := AlertingRule{}
+		alertingRule.Type = string(RuleTypeAlerting)
 		if err := json.Unmarshal(rule, &alertingRule); err == nil {
```

Comment:
So first of all this code was wrong, we don't even capture error anywhere no? How you got that `type field not present in rule` if it's not propagated here?

Reply:
I think we are not justing the UnmarshalJSON function at all now that I see, we should be using UnMarshalJSON instead of json.UnMarshal what do you think?

Reply:
I think changing the commentary on the Rules might be a better way since then we have a clear way of checking how the type field exists which is not clear right now. But you know the code better than me so happy to comply with what you think :)

Reply:
again, let's add unit test for your "broken" case and we will go from there (: